### PR TITLE
Give Rover Sprayer and Gripper support

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -727,6 +727,25 @@ def wait_distance(self, distance, accuracy=5, timeout=30):
         self.progress("Failed to attain distance %u" % distance)
         raise WaitDistanceTimeout()
 
+    def wait_servo_channel_value(self, channel, value, timeout=2):
+        '''wait for channel to hit value'''
```
Comment (khancyr) [inline Tools/autotest/common.py]: """"

File: ArduCopter/ArduCopter.cpp [hunk 2]
```diff
@@ -412,6 +415,8 @@ void Copter::three_hz_loop()
     sprayer.update();
 #endif
 
+    update_events();
```
Comment (khancyr) [inline ArduCopter/ArduCopter.cpp]: why do you need this one in addition to the scheduled one ?

File: libraries/RC_Channel/RC_Channel.cpp [hunk 3]
```diff
@@ -473,13 +480,51 @@ void RC_Channel::do_aux_function_relay(const uint8_t relay, bool val)
...
+}
+
 void RC_Channel::do_aux_function(const aux_func_t ch_option, const aux_switch_pos_t ch_flag)
 {
     switch(ch_option) {
     case CAMERA_TRIGGER:
         do_aux_function_camera_trigger(ch_flag);
         break;
 
+    case GRIPPER:
```
Comment (amilcarlucas) [inline libraries/RC_Channel/RC_Channel.cpp]: How about keeping the code order: relay, sprayer, gripper

File: libraries/RC_Channel/RC_Channel.h [hunk 4]
```diff
@@ -179,8 +179,10 @@ class RC_Channel {
     virtual void init_aux_function(aux_func_t ch_option, aux_switch_pos_t);
     virtual void do_aux_function(aux_func_t ch_option, aux_switch_pos_t);
 
-    void do_aux_function_relay(uint8_t relay, bool val);
     void do_aux_function_camera_trigger(const aux_switch_pos_t ch_flag);
+    void do_aux_function_gripper(const aux_switch_pos_t ch_flag);
```
Comment (amilcarlucas) [inline libraries/RC_Channel/RC_Channel.h]: How about keeping the code order: relay, sprayer, gripper

**Comment by peterbarker** [comment]
On Wed, 15 Aug 2018, Amilcar Lucas wrote:

> How about keeping the code order: relay, sprayer, gripper

I'm going for alphabetical order :-)

**Comment by amilcarlucas** [comment]
Or that. :)
But try to make it consistent, ind the function declarations, definitions and case statements.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**